packages/core/src/domain/session/sessionStore.ts

```diff
@@ -65,11 +66,17 @@ export function startSessionStore<TrackingType extends string>(
   const watchSessionTimeoutId = setInterval(watchSession, STORAGE_POLL_DELAY)
   let sessionCache: SessionState = retrieveActiveSession()
 
+  initializeSession()
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏it is a bit confusing to me that `retrieveActiveSession` and `initializeSession` are both accessing the store and having logic around the initial state.
Could we init `sessionCache` with the `initialSessionState` and let `initializeSession` handling all the store accesses and initial state logic ?